pkg/kotsadm/main.go

```diff
@@ -168,6 +166,40 @@ func Deploy(deployOptions types.DeployOptions) error {
 	return nil
 }
 
+func canUpgrade(upgradeOptions types.UpgradeOptions, clientset *kubernetes.Clientset, log *logger.Logger) error {
+	_, err := clientset.CoreV1().Namespaces().Get(upgradeOptions.Namespace, metav1.GetOptions{})
+	if kuberneteserrors.IsNotFound(err) {
+		err := errors.New("The namespace cannot be found or accessed")
+		return err
+	}
+
+	if upgradeOptions.ForceUpgradeKurl {
+		return nil
+	}
+
+	// don't upgrade kurl clusters.  kurl only installs into default namespace.
+	if upgradeOptions.Namespace != "" && upgradeOptions.Namespace != "default" {
+		return nil
+	}
+
+	kotsAPI, err := clientset.AppsV1().Deployments(upgradeOptions.Namespace).Get("kotsadm-api", metav1.GetOptions{})
```

Comment:
I think we can ignore from here down in the function. Why should we check that env var?

Reply:
@marccampbell because kotsamd-api can exist and not be installed with kurl?  isn't the only way to test that is to test the `ENABLE_KURL` variable?